x/mongo/driver/topology/server.go

```diff
@@ -605,8 +657,7 @@ func (s *Server) update() {
 
 		previousDescription := s.Description()
 
-		// Perform the next check.
-		desc, err := s.check()
+		desc, err := checkServerWithSignal(s, s.conn, s.heartbeatListener)
 		if errors.Is(err, errCheckCancelled) {
```

Comment:
We should do something with this error if it's not `errCheckCancelled`. What should we do if there's some other network or server error?

Reply:
We never did anything with it beforehand. Do you have any suggestions?

Reply:
Looking at the code for `checkServerWithSignal` and `server.check`, I realize most errors are passed through `desc.LastError`. Seems like it's working as designed, but it's difficult to follow and may be a source of bugs in the future. For now, let's defer improving it to a future refactor.

Optional: Consider adding a comment that mentions that the only error returned from `checkServerWithSignal` is `errCheckCancelled`.